src/Entity/Membership.php

```diff
@@ -200,6 +200,11 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
       ->setDisplayConfigurable('view', TRUE)
       ->setRevisionable(TRUE)
       ->setSetting('workflow_callback', ['\Drupal\membership\Entity\Membership', 'getWorkflowId']);
+    $fields['provider'] = BaseFieldDefinition::create('membership_provider_id')
```

Comment:
So this isn't the configuration for the provider - it's a reference to a provider ID and an ID for the provider to parse. E.g., it would be `membership_term` and, say, `34`, which would maybe be interpreted by your module as Membership Term 34. Or, maybe, nothing in the ID portion and this just says to ask the `membership_term` plugin to configure itself for the membership. I'll add more documentation.